src/Compilers/CSharp/Test/Semantic/Semantics/SpanStackSafetyTests.cs

```diff
@@ -21,6 +15,264 @@ namespace Microsoft.CodeAnalysis.CSharp.UnitTests
     public class SpanStackSafetyTests : CompilingTestBase
     {
         [Fact]
+        public void SpanToObjectPatternSwitch()
+        {
+            var comp = CreateCompilationWithMscorlibAndSpan(@"
+using System;
+
+class C
+{
+    void M(object o)
+    {
+        switch (o)
+        {
+            case Span<int> span:
+                break;
+            default:
+                break;
+        }
+    }
+}");
+            comp.VerifyDiagnostics(
+                // (10,18): error CS8121: An expression of type 'object' cannot be handled by a pattern of type 'Span<int>'.
+                //             case Span<int> span:
+                Diagnostic(ErrorCode.ERR_PatternWrongType, "Span<int>").WithArguments("object", "System.Span<int>").WithLocation(10, 18));
+        }
+
+        [Fact]
+        public void SpanToGenericPatternSwitch()
+        {
+            var comp = CreateCompilationWithMscorlibAndSpan(@"
+using System;
+class C
+{
+    void M<T>(T t)
+    {
+        switch (t)
+        {
+            case Span<int> span:
+                break;
+            default:
+                break;
+        }
+    }
+}");
+            comp.VerifyDiagnostics(
+                // (9,18): error CS8121: An expression of type 'T' cannot be handled by a pattern of type 'Span<int>'.
+                //             case Span<int> span:
+                Diagnostic(ErrorCode.ERR_PatternWrongType, "Span<int>").WithArguments("T", "System.Span<int>").WithLocation(9, 18));
+        }
+
+        [Fact]
+        public void SpanToGenericPatternSwitch2()
+        {
+            var comp = CreateCompilationWithMscorlibAndSpan(@"
+using System;
+class C
+{
+    void M<T>(T t) where T : struct
+    {
+        switch (t)
+        {
+            case Span<int> span:
+                break;
+            default:
+                break;
+        }
+    }
+}");
+            comp.VerifyDiagnostics(
+                // (9,18): error CS8121: An expression of type 'T' cannot be handled by a pattern of type 'Span<int>'.
+                //             case Span<int> span:
+                Diagnostic(ErrorCode.ERR_PatternWrongType, "Span<int>").WithArguments("T", "System.Span<int>").WithLocation(9, 18));
+        }
+
+        [Fact]
+        public void ObjectToSpanPatternSwitch()
+        {
+            var comp = CreateCompilationWithMscorlibAndSpan(@"
+using System;
+class C
+{
+    void M(Span<string> s)
+    {
+        switch (s)
+        {
+            case Span<object> span:
```

Comment:
What about an identity case? Dont see that tested in this PR but seems like it would be valid

Reply:
It should be valid and I'll add a test for it.